### PR TITLE
Add CI workflow to run unit tests with latest from Woodwork main

### LINKED ISSUE #1926: Add CI workflow to run unit tests with latest from Woodwork main
- We should add a CI workflow that does the following:

1. Install Featuretools 
2. Install Woodwork from the `main` branch
3. Verify that the Featuretools unit tests run

This workflow should run on the following:
```yaml
on:
  schedule:
    - cron: '0 * * * *'
  pull_request:
    types: [opened, synchronize]
  push:
    branches:
      - main
  workflow_dispatch:
```

### PR DESCRIPTION
**Pull request by mingdavidqi**
- Fixes #1926

### CODE REVIEW

File: .github/workflows/test_with_woodwork_main_branch.yml [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+      - main
+  workflow_dispatch:
+jobs:
+  scheduled_unit_latest_tests:
+    name: ${{ matrix.python_version }} unit tests ${{ matrix.libraries }}
+    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: true
+      matrix:
+        python_version: ["3.8", "3.9", "3.10"]
```
Comment (thehomebrewnerd) [inline .github/workflows/test_with_woodwork_main_branch.yml:18]: Featuretools doesn't support `3.10` yet, but we still do support `3.7`, so It think this list should be `["3.7", "3.8", "3.9"]` instead.

File: docs/source/release_notes.rst [hunk 2]
```diff
@@ -19,9 +19,10 @@ Future Release
     * Testing Changes
         * Update test cases to cover __main__.py file (:pr:`1927`)
         * Pin ``responses`` version to be below 0.19.0 in minimum_test_requirements to avoid moto error (:pr:`1929`)
+        * Add CI workflow to test with latest woodwork main branch (:pr:`1936`)
 
     Thanks to the following people for contributing to this release:
-    :user:`tamargrey`, :user:`kushal-gopal`, :user:`rwedge`, :user:`mingdavidqi`, :user:`andriyor`
+    :user:`tamargrey`, :user:`kushal-gopal`, :user:`rwedge`, :user:`mingdavidqi`, :user:`andriyor`, :user:`mingdavidqi`
```
Comment (thehomebrewnerd) [inline docs/source/release_notes.rst:25]: Your username is already in the list - don't need to add it again.

File: koalas-requirements.txt [hunk 3]
```diff
@@ -1,4 +1,4 @@
 pyspark>=3.0.0,<3.2.0
-koalas>=1.8.1,<=1.8.2
+koalas>=1.8.0,<=1.8.2; python_version < '3.10'
```
Comment (thehomebrewnerd) [inline koalas-requirements.txt:2]: Not necessary since FT doesn't support 3.10 yet.

File: .github/workflows/test_with_woodwork_main_branch.yml [hunk 4]
```diff
@@ -0,0 +1,61 @@
+name: Test Featuretools with main Branch
```
Comment (thehomebrewnerd) [inline .github/workflows/test_with_woodwork_main_branch.yml:1]: Add Woodwork to this -> `Test Featuretools with Woodwork main branch`

**Review by thehomebrewnerd — COMMENTED** [review]
Left a couple of comments, but also have a question about the intended behavior here.

@gsheni Are we expecting this series of checks to run as part of every new PR to Featuretools or do we want this to be a scheduled run that periodically runs the test suite with WW main and alerts us on a failure?

File: .github/workflows/test_with_woodwork_main_branch.yml (hunk 1 above)
Comment (mingdavidqi) [reply .github/workflows/test_with_woodwork_main_branch.yml]: updated

File: docs/source/release_notes.rst (hunk 2 above)
Comment (mingdavidqi) [reply docs/source/release_notes.rst]: updated

File: koalas-requirements.txt (hunk 3 above)
Comment (mingdavidqi) [reply koalas-requirements.txt]: updated

File: .github/workflows/test_with_woodwork_main_branch.yml (hunk 4 above)
Comment (mingdavidqi) [reply .github/workflows/test_with_woodwork_main_branch.yml]: updated

File: .github/workflows/test_with_woodwork_main_branch.yml [hunk 5]
```diff
@@ -0,0 +1,61 @@
+name: Test Featuretools with Woodwork main Branch
+on:
+  schedule:
+    - cron: "0 * * * *"
```
Comment (gsheni) [inline .github/workflows/test_with_woodwork_main_branch.yml:4]: Won't this be excessive? 
@rwedge Should we lower this or come up with another trigger?

Comment (rwedge) [reply .github/workflows/test_with_woodwork_main_branch.yml:4]: @gsheni if we fix the issue with https://github.com/alteryx/gh-action-ci we could have it only actually run if there's been a recent commit

File: .github/workflows/test_with_woodwork_main_branch.yml [hunk 6]
```diff
@@ -0,0 +1,61 @@
...
+      - main
+  workflow_dispatch:
+jobs:
+  scheduled_unit_latest_tests:
+    name: ${{ matrix.python_version }} unit tests ${{ matrix.libraries }}
+    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: true
+      matrix:
+        python_version: ["3.7", "3.8", "3.9",]
```
Comment (rwedge) [inline .github/workflows/test_with_woodwork_main_branch.yml:18]: "3.10" should be added, but we'll want to exclude the "3.10 koalas"  -- see this discussion in the PR that added 3.10 tests

https://github.com/alteryx/featuretools/pull/1940#discussion_r823131888

File: .github/workflows/test_with_woodwork_main_branch.yml [hunk 7]
```diff
@@ -0,0 +1,61 @@
...
+        libraries: ["core", "dask", ]
```
Comment (rwedge) [inline .github/workflows/test_with_woodwork_main_branch.yml:19]: "core" and "koalas"

Comment (mingdavidqi) [reply .github/workflows/test_with_woodwork_main_branch.yml]: Added koalas and removed dask

File: .github/workflows/test_with_woodwork_main_branch.yml (hunk 6 above)
Comment (mingdavidqi) [reply .github/workflows/test_with_woodwork_main_branch.yml]: Added 3.10. Using if condition to exclude the "3.10 koalas". Syntax of exclude is working weird when I tested with act locally.

File: .github/workflows/test_with_woodwork_main_branch.yml [hunk 8]
```diff
@@ -0,0 +1,59 @@
...
+          pip config --site set global.progress_bar off
+          python -m pip install -U pip
+          sudo apt update && sudo apt install -y graphviz
+
+      - if: ${{ matrix.libraries == 'core' }}
+        name: Install Woodwork & Featuretools - test requirements
+        run: |
+          python -m pip install -e unpacked_sdist/[test]
+          python -m pip install -U git+https://github.com/alteryx/woodwork
+      - if: ${{ matrix.libraries == 'dask' }}
```
Comment (rwedge) [inline .github/workflows/test_with_woodwork_main_branch.yml:40]: remove dask since not in the job matrix

File: docs/source/release_notes.rst [hunk 9]
```diff
@@ -21,9 +21,11 @@ Future Release
         * Update test cases to cover __main__.py file (:pr:`1927`)
         * Upgrade moto requirement (:pr:`1929`, :pr:`1938`)
         * Add Python 3.9 linting, install complete, and docs build CI tests (:pr:`1934`)
+        * Add CI workflow to test with latest woodwork main branch (:pr:`1936`)
         * Add support for Python 3.10 (:pr:`1940`)
         * Add lower bound for wheel for minimum dependency checker and limit lint CI tests to Python 3.10 (:pr:`1945`)
 
+
```
Comment (rwedge) [inline docs/source/release_notes.rst:28]: remove extra line

File: .github/workflows/test_with_woodwork_main_branch.yml [hunk 10]
```diff
@@ -0,0 +1,59 @@
+name: Test Featuretools with Woodwork main Branch
+on:
+  schedule:
+    - cron: '0 * * * *'
+  pull_request:
+    types: [opened, synchronize]
+  push:
+    branches:
+      - main
```
Comment (rwedge) [inline .github/workflows/test_with_woodwork_main_branch.yml:9]: let's only do schedule and workflow dispatch for now

File: .github/workflows/test_with_woodwork_main_branch.yml [hunk 11]
```diff
@@ -0,0 +1,59 @@
+name: Test Featuretools with Woodwork main Branch
+on:
+  schedule:
+    - cron: '0 * * * *'
```
Comment (rwedge) [inline .github/workflows/test_with_woodwork_main_branch.yml:4]:
```suggestion
    - cron: '0 14 * * *'
```

File: .github/workflows/test_with_woodwork_main_branch.yml [hunk 12]
```diff
@@ -0,0 +1,59 @@
...
+          python -m pip install -U git+https://github.com/alteryx/woodwork dask[dataframe]>=2021.10.0
+      - if: ${{ matrix.libraries == 'koalas' && matrix.python_version != '3.10' }}
+        name: Install Woodwork & Featuretools with koalas pkg - koalas requirements
+        run: |
+          sudo apt install -y openjdk-11-jre-headless
+          JAVA_HOME="/usr/lib/jvm/java-11-openjdk-amd64"
+          PYSPARK_SUBMIT_ARGS="--master local[2] pyspark-shell"
+          python -m pip install -e unpacked_sdist/[koalas]
+          python -m pip install -e unpacked_sdist/[test]
+          python -m pip install -U git+https://github.com/alteryx/woodwork
```
Comment (rwedge) [inline .github/workflows/test_with_woodwork_main_branch.yml:53]: This command is shared with the "core" install

Assuming the exclude change works, we could make these two lines be a shared step, with the earlier lines of this step becoming a koalas-only step that occurs before the shared step

File: .github/workflows/test_with_woodwork_main_branch.yml [hunk 13]
```diff
@@ -0,0 +1,61 @@
...
+  workflow_dispatch:
+jobs:
+  scheduled_unit_latest_tests:
+    name: ${{ matrix.python_version }} unit tests ${{ matrix.libraries }}
+    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: true
+      matrix:
+        python_version: ["3.7", "3.8", "3.9", "3.10"]
+        libraries: ["core", "koalas", ]
```
[the remark this answers is not in the document]
Comment (mingdavidqi) [reply .github/workflows/test_with_woodwork_main_branch.yml:14]: updated

File: .github/workflows/test_with_woodwork_main_branch.yml (hunk 8 above)
Comment (mingdavidqi) [reply .github/workflows/test_with_woodwork_main_branch.yml]: removed

File: docs/source/release_notes.rst (hunk 9 above)
Comment (mingdavidqi) [reply docs/source/release_notes.rst]: removed

File: .github/workflows/test_with_woodwork_main_branch.yml (hunk 10 above)
Comment (mingdavidqi) [reply .github/workflows/test_with_woodwork_main_branch.yml]: updated

File: .github/workflows/test_with_woodwork_main_branch.yml (hunk 11 above)
Comment (mingdavidqi) [reply .github/workflows/test_with_woodwork_main_branch.yml]: updated

File: .github/workflows/test_with_woodwork_main_branch.yml (hunk 12 above)
Comment (mingdavidqi) [reply .github/workflows/test_with_woodwork_main_branch.yml:42]: updated

**Review by rwedge — APPROVED** [review]
Looks good!

**Pull request merged**